app/controllers/admin/products_controller.rb

```diff
@@ -1,24 +1,19 @@
 class Admin::ProductsController < ApplicationController
-  before_action :find_product, only: [:show, :edit, :update, :destroy]
+  before_action :find_product, only: [:new, :create, :show, :edit, :update, :destroy]
 
   def index
     @products = Product.all
   end
 
   def new
-    @product = Product.new
-    @product_image = @product.product_images.build
+    @product_form = ProductForm.new(@product)
```

Comment:
Oh maybe we could explore refactoring this line and line 27 into the same thing. Then lines 14 and 32 into the same as well.